src/main/java/com/blackducksoftware/integration/hub/jenkins/scan/BDCommonScanStep.java

```diff
@@ -383,14 +378,26 @@ private ProjectView getProjectFromVersion(final ProjectRequestService projectReq
         return projectVersion;
     }
 
-    private ProjectVersionView getProjectVersionFromScanStatus(final CodeLocationRequestService codeLocationRequestService,
-            final ProjectVersionRequestService projectVersionRequestService, final MetaService metaService, final ScanSummaryView scanSummaryItem)
-            throws IntegrationException {
-        final CodeLocationView codeLocationItem = codeLocationRequestService
-                .getItem(metaService.getFirstLink(scanSummaryItem, MetaService.CODE_LOCATION_BOM_STATUS_LINK), CodeLocationView.class);
-        final String projectVersionUrl = codeLocationItem.mappedProjectVersion;
-        final ProjectVersionView projectVersion = projectVersionRequestService.getItem(projectVersionUrl, ProjectVersionView.class);
-        return projectVersion;
+    private boolean isShouldWaitForScansFinished(final Run<?, ?> run) {
+        if (!isDryRun()) {
+            if (isShouldGenerateHubReport()) {
+                return true;
+            }
+            final Object parent = run.getParent();
+            if (parent instanceof FreeStyleProject) {
```

Comment:
If we move this code up into  the calling classes, we would know the Parent class type so we can remove one of these if's. We can then pass into the constructor if the Failure Condition is configured or not, and this method would reduce to 
`return !isDryRun() && (isShouldGenerateHubReport() || isFailureConditionConfigured());`